Kubewarden v1.5.0 release post. (#167)
kubewarden/kubewarden.io: content/blog/2023/01/release-1_5_0.md

---
title: Kubewarden 1.5.0 release
authors:
- José Guilherme Vanz
date: 2023-01-18
---

Today we're pleased to announce the availability of Kubewarden 1.5.0!

Among all the dependencies updates and minor fixes in all the Kubewarden components it's worth
to mention some of them and a new interesting feature.

## Policy evaluation timeout

> In the continuous effort of the Kubewarden team to keep the project safer on every release,
> policy server now allows users to define how much time a policy can spend to evaluate an admission request.
> This security feature, which is enabled by default, ensures that infinite loops and other scenarios
> where policies may get stuck will not consume all the resources available turning policy server more
> reliable to Denial of Service (DOS) attacks.

Reviewer: ```suggestion
The Kubewarden team is constantly working to improve the security posture of the project. As part of this efforts,
we're excited to introduce the new "policy evaluation timeout" feature.

Starting from this release, Policy Server will interrupt the evaluation of admission requests after a certain amount of time has elapsed.
This security feature, which is enabled by default, prevents a Policy Server from running out of computing
resources because one or more of its policy evaluations are stuck in infinite loops.
This mitigates a type of Denial Of Service (DOS) attacks against the Policy Server. 
```

Starting from this release, Policy Server will interrupt the evaluation of admission requests after a certain amount of time has elapsed.
This security feature, which is enabled by default, prevents a Policy Server from running out of computing
resources because one or more of its policy evaluations are stuck in infinite loops.
This mitigates a type of Denial Of Service (DOS) attacks against the Policy Server.

Take a look in the Kubewarden documentation to get more information about this new feature.

## Changes to kwctl

`kwctl` now imports the trusted certificate authorities from from the host system. Thus, `kwctl`
interactions with registries secured with certificates issued by 3rd party certificate authorities
becomes simpler.
Note, it's still possible to fine tune the certificates to be used via the [`sources.yml` file](https://docs.kubewarden.io/distributing-policies/custom-certificate-authorities#the-sourcesyaml-file).

## Changes to Kubewarden controller

The validation of `ClusterAdmissionPolicy` and `AdmissionPolicy` custom resources has been extended. The
new validation checks are able to prevent the creation of resources that do not have any `rule` specified.

> We would like to thank [Mauren Berti ](https://github.com/stormqueen1990) for this contribution.

Reviewer: ```suggestion
We would like to thank [Mauren Berti](https://github.com/stormqueen1990) for this contribution.
```

## Miscellaneous

Furthermore, the changelog of the main Kubewarden components (policy-server, kubewarden-controller and kwctl) are now being generated using the [release-drafter](https://github.com/release-drafter/release-drafter) project.

> Starting from this release, the Changelog file is no longer available inside of the root of these git projects, rather

Reviewer: ```suggestion
Starting from this release, the `Changelog` file is no longer available inside of the root of these git projects, rather
```

---
title: Kubewarden 1.5.0 release
authors:
- José Guilherme Vanz
date: 2023-01-18
---

Today we're pleased to announce the availability of Kubewarden 1.5.0!

This release brings the usual amount of small bug fixes, dependency updates and a major security feature.
Let's take a closer look!

## Policy evaluation timeout

The Kubewarden team is constantly working to improve the security posture of the project. As part of this efforts,
we're excited to introduce the new "policy evaluation timeout" feature.

> Starting from this release, Policy Server will interrupt the evaluation of admission requests after a certain amount of time has elapsed.

Reviewer 2: This needs to be phrased better IMO. Certain amount of time has elapsed after what? I understand it prevents the Policy Server from DOS attacks by preventing it from infinite loops and that we explain this context in the next sentence. But we need to make it clear. I'd recommend we set the context first by stating the problem first a.k.a. talk about potential DOS attacks or running out of compute resources on account of infinite loops. Then present the solution i.e. the Policy Server interrupting the evaluation of admission requests

This security feature, which is enabled by default, prevents a Policy Server from running out of computing
resources because one or more of its policy evaluations are stuck in infinite loops.
This mitigates a type of Denial Of Service (DOS) attacks against the Policy Server.

> Take a look in the Kubewarden documentation to get more information about this new feature.

Reviewer 2: ```suggestion
Take a look at the Kubewarden documentation to get more information about this new feature.
```

## Changes to kwctl

`kwctl` now imports the trusted certificate authorities from from the host system. Thus, `kwctl`
interactions with registries secured by certificates issued by 3rd party certificate authorities
becomes simpler.

Note, it's still possible to fine tune the certificates to be used via the [`sources.yml` file](https://docs.kubewarden.io/distributing-policies/custom-certificate-authorities#the-sourcesyaml-file).

## Changes to Kubewarden controller

The validation of `ClusterAdmissionPolicy` and `AdmissionPolicy` custom resources has been extended. The
new validation checks are able to prevent the creation of resources that do not have any `rule` specified.

> We would like to thank [Mauren Berti](https://github.com/stormqueen1990) for this contribution.

Reviewer 2: Mauren's a friend and I mean her no harm, but she is an employee of SUSE now. I'm not sure if we should be singling out folks unless they're community members like this.

Author: @stormqueen1990, are you okay being cited here ?

Reviewer 3: I share @divya-mohan0209's concern. I'd prefer not being nominally cited.

Author: I agree with @divya-mohan0209. I've removed the link

## Miscellaneous

Furthermore, the changelog of the main Kubewarden components (policy-server, kubewarden-controller and kwctl) are now being generated using the [release-drafter](https://github.com/release-drafter/release-drafter) project.

Starting from this release, the `Changelog` file is no longer available inside of the root of these git projects, rather

> is made available under the GitHub Release pages.

Reviewer 2: I assume, we mean that the Changelog has now shifted like this - https://github.com/kubewarden/kubewarden-controller/releases/tag/v1.5.0
```suggestion
it has been made available under the release section of the individual repositories.
```

> ## Go grab it!

Reviewer 2: ```suggestion
## Go, grab it!
```

> We are eager to know what do you think about Kubewarden and this release.

Reviewer 2: ```suggestion
We are eager to know what you think about Kubewarden and this release.
```

> As usual, you can reach out to us over our slack channel.

Reviewer 2: ```suggestion
Reach out to us over our slack channel or join one of our [monthly community meetings](https://www.kubewarden.io/blog/2022/12/community-meeting/) to know more.
```
